### PR TITLE
perf(s3stream): optimize StreamRecordBatchList to reduce object creation

### PR DESCRIPTION
**Pull request by lifepuzzlefun**
1. remove ArrayList copy when create instance.
2. StreamRecordBatch will implement the ComparableItem interface without create wrap object.

### CODE REVIEW

**Comment by daniel-y** [comment]
Hi @lifepuzzlefun thanks for your contribution. And, please check the failed actions.

**Comment by SCNieh** [comment]
LGTM, @lifepuzzlefun the commit msg lint seems failed, maybe rebase & rewrite the commit msg to something like:  perf(s3stream): optimize StreamRecordBatchList
- xxx

File: s3stream/src/main/java/com/automq/stream/utils/biniarysearch/primitive/LongOrderedCollection.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.automq.stream.utils.biniarysearch.primitive;
+
+public abstract class LongOrderedCollection {
```
Comment (daniel-y) [inline s3stream/src/main/java/com/automq/stream/utils/biniarysearch/primitive/LongOrderedCollection.java:19]: It's duplicated with AbstractOrderedCollection, maintain two similar code segments is too boring IMO. Do we really need pay attention to the box/unbox issues?

Comment (lifepuzzlefun) [reply s3stream/src/main/java/com/automq/stream/utils/biniarysearch/primitive/LongOrderedCollection.java]: agree. maybe we can create primitive type only we this become a issue. the relate code reverted for now. please take a look.

**Comment by lifepuzzlefun** [comment]
@SCNieh @Chillax-0v0 thanks for help. the commit message has been modified, hope the CI can pass. by the way, may be we need guide the contributor write commit message in the expected format.

**Comment by zhouxinyu** [comment]
We will merge the PR in a squash manner, so we don't need to modify the commit message one by one, just squashing all the commits to one commit is enough.

Anyway, I will remove the `Lint Commit Messages` action, since we don't need it anymore, sorry for your time. @lifepuzzlefun

**Pull request merged**